### PR TITLE
chore: bump TSTyche to v2

### PR DESCRIPTION
**Pull request by mrazauskas**
This PR upgrades TSTyche to v2, which was published recently.

The new version ships with the watch mode and renamed / improved matchers. The detailed release notes can be found here: https://tstyche.org/releases/tstyche-2

I have updated the type tests with the new matcher names accordingly.

### CODE REVIEW

File: packages/ast-client/src/query.test-d.ts [hunk 1]
```diff
@@ -38,65 +38,65 @@ const emptyRecordQuery = {} as Query<Record<string, never>, MyQueryResult>;
...
-    expect<DocumentNode>().type.toBeAssignable(myQuery);
-    expect<DocumentNode>().type.toBeAssignable(voidQuery);
-    expect<DocumentNode>().type.toBeAssignable(undefinedQuery);
-    expect<DocumentNode>().type.toBeAssignable(emptyQuery);
-    expect<DocumentNode>().type.toBeAssignable(emptyRecordQuery);
+    expect(myQuery).type.toBeAssignableTo<DocumentNode>();
+    expect(voidQuery).type.toBeAssignableTo<DocumentNode>();
+    expect(undefinedQuery).type.toBeAssignableTo<DocumentNode>();
+    expect(emptyQuery).type.toBeAssignableTo<DocumentNode>();
+    expect(emptyRecordQuery).type.toBeAssignableTo<DocumentNode>();
```
Comment (mrazauskas) [inline packages/ast-client/src/query.test-d.ts:53]: Source and target got swapped in these assertions. The new `.toBeAssignableTo()` matcher now allows keeping the target type on right side similar to other cases.

I understood that `.toBeAssignableTo()` is missing while looking through your repository. Thanks for using TSTyche (;

File: package.json [hunk 2]
```diff
@@ -19,7 +19,8 @@
     "format": "prettier -w .",
     "format:check": "prettier -c .",
     "prepack": "pnpm run dist",
-    "test": "pnpm -r test"
+    "test": "pnpm -r test",
+    "test:types": "pnpm -r test:types"
```
Comment (mrazauskas) [inline package.json:23]: Just an idea.

**Comment by m4rw3r** [comment]
Thank you so much for this! 👍

**Comment by mrazauskas** [comment]
Welcome! I just wanted to make sure that all works as expected. Thanks for using TSTyche.

**Pull request merged**